stackslib/src/chainstate/stacks/boot/signers_voting_tests.rs

```diff
@@ -364,13 +361,14 @@ fn vote_for_aggregate_public_key_in_last_block() {
     let receipts = block.receipts.as_slice();
     assert_eq!(receipts.len(), 1);
 
-    // vote should succeed
+    // vote fails because the reward cycle has changed
+    //  and the signer set hasn't been set yet.
```

Comment:
How can the reward cycle change during a tenure?

Reply:
"Bobs block" in this test is the first block of the next reward cycle (it's after the prepare phase ended).

Reply:
It is in the same tenure as Alice txs

Reply:
🤔

Intriguing. I added some print statements to see what's going on:

```
INFO [1706824307.469257] [clarity/./src/vm/functions/mod.rs:620] [chainstate::stacks::boot::signers_voting_tests::vote_for_aggregate_public_key_in_last_block] (tuple (burn-height u39) (cycle u8) (event "test-1") (signer-index u0))
INFO [1706824307.471293] [stackslib/src/chainstate/stacks/db/transactions.rs:1007] [chainstate::stacks::boot::signers_voting_tests::vote_for_aggregate_public_key_in_last_block] Contract-call successfully processed, contract_name: ST000000000000000000002AMW42H.signers-voting, function_name: vote-for-aggregate-public-key, function_args: [u0, 0x020000000000000000000000000000000000000000000000000000000000000000, u0], return_value: (err u10006), cost: ExecutionCost { write_length: 88, write_count: 1, read_length: 11240, read_count: 19, runtime: 72890 }
...
INFO [1706824307.487501] [clarity/./src/vm/functions/mod.rs:620] [chainstate::stacks::boot::signers_voting_tests::vote_for_aggregate_public_key_in_last_block] (tuple (burn-height u40) (cycle u9) (event "test-1") (signer-index u1))
INFO [1706824307.488056] [stackslib/src/chainstate/stacks/db/transactions.rs:1007] [chainstate::stacks::boot::signers_voting_tests::vote_for_aggregate_public_key_in_last_block] Contract-call successfully processed, contract_name: ST000000000000000000002AMW42H.signers-voting, function_name: vote-for-aggregate-public-key, function_args: [u1, 0x020000000000000000000000000000000000000000000000000000000000000000, u0], return_value: (err u2), cost: ExecutionCost { write_length: 0, write_count: 0, read_length: 8544, read_count: 10, runtime: 33662 }
```

In alice's transaction, burn-block-height is 39, in bob's its 40. So yeah, burn-block-height is going up.

Will need to see if this is a bug in `TestPeer` or in the actual handling of burn block heights.

Reply:
I think this is a bug (or unexpected feature!) in the nakamoto burn state handling in the Clarity VM -- the tip it uses is the traditional one used in Stacks: the burn block of its parent block. This means that the first block of the tenure and the second block of the tenure will have different burn views. Will open an issue for this because its not really related to this PR.